Adapted version of

commit 836846f0d747e1be8e37d2d43b215a68b30ea1a9
Author: Lorenzo Colitti <lorenzo@google.com>
Date:   Thu Mar 16 12:54:20 2017 +0900

    iptables: move XT_LOCK_NAME from CFLAGS to config.h.
    
    This slightly simplifies configure.ac and results in more
    correct dependencies.
    
    Tested by running ./configure with --with-xt-lock-name and
    without, and using strace to verify that the right lock is used.
    
    $ make distclean-recursive && ./autogen.sh &&
      ./configure --disable-nftables --prefix /tmp/iptables &&
      make -j64 &&
      make install &&
      sudo strace -e open,flock /tmp/iptables/sbin/iptables -L foo
    ...
    open("/run/xtables.lock", O_RDONLY|O_CREAT, 0600) = 3
    flock(3, LOCK_EX|LOCK_NB)               = 0
    
    $ make distclean-recursive && ./autogen.sh && \
      ./configure --disable-nftables --prefix /tmp/iptables \
      --with-xt-lock-name=/tmp/iptables/run/xtables.lock &&
      make -j64 &&
      make install &&
      sudo strace -e open,flock /tmp/iptables/sbin/iptables -L foo
    ...
    open("/tmp/iptables/run/xtables.lock", O_RDONLY|O_CREAT, 0600) = 3
    flock(3, LOCK_EX|LOCK_NB)               = 0
    
    Signed-off-by: Lorenzo Colitti <lorenzo@google.com>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

diff -up iptables-1.4.21/configure.ac.move_XT_LOCK_NAME_to_config.h iptables-1.4.21/configure.ac
--- iptables-1.4.21/configure.ac.move_XT_LOCK_NAME_to_config.h	2017-04-05 14:48:11.855229929 +0200
+++ iptables-1.4.21/configure.ac	2017-04-05 14:48:11.856229937 +0200
@@ -122,7 +122,7 @@ AM_CONDITIONAL([HAVE_LIBNFNETLINK], [tes
 regular_CFLAGS="-Wall -Waggregate-return -Wmissing-declarations \
 	-Wmissing-prototypes -Wredundant-decls -Wshadow -Wstrict-prototypes \
 	-Winline -pipe";
-regular_CPPFLAGS="${largefile_cppflags} -DXT_LOCK_NAME=\\\"\${xt_lock_name}\\\" -D_REENTRANT \
+regular_CPPFLAGS="${largefile_cppflags} -D_REENTRANT \
 	-DXTABLES_LIBDIR=\\\"\${xtlibdir}\\\" -DXTABLES_INTERNAL";
 kinclude_CPPFLAGS="";
 if [[ -n "$kbuilddir" ]]; then
@@ -160,7 +160,9 @@ AC_SUBST([libxtables_vcurrent])
 AC_SUBST([libxtables_vage])
 libxtables_vmajor=$(($libxtables_vcurrent - $libxtables_vage));
 AC_SUBST([libxtables_vmajor])
-AC_SUBST([xt_lock_name])
+
+AC_DEFINE_UNQUOTED([XT_LOCK_NAME], "${xt_lock_name}",
+	[Location of the iptables lock file])
 
 AC_CONFIG_FILES([Makefile extensions/GNUmakefile include/Makefile
 	iptables/Makefile iptables/xtables.pc
diff -up iptables-1.4.21/iptables/xshared.c.move_XT_LOCK_NAME_to_config.h iptables-1.4.21/iptables/xshared.c
--- iptables-1.4.21/iptables/xshared.c.move_XT_LOCK_NAME_to_config.h	2017-04-05 14:48:11.855229929 +0200
+++ iptables-1.4.21/iptables/xshared.c	2017-04-05 14:48:11.856229937 +0200
@@ -1,3 +1,4 @@
+#include <config.h>
 #include <getopt.h>
 #include <errno.h>
 #include <libgen.h>
